### ISSUE TITLE
No package kernel-devel-4.18.7-1.el7.elrepo.x86_64 available

### ISSUE THREAD

**Issue by zwl1619** [dotless-de/vagrant-vbguest#307]
```
λ vagrant reload
==> default: Attempting graceful shutdown of VM...
==> default: Clearing any previously set forwarded ports...
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
    default: Adapter 2: hostonly
==> default: Forwarding ports...
    default: 8000 (guest) => 8000 (host) (adapter 1)
    default: 22 (guest) => 2222 (host) (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: root
    default: SSH auth method: password
==> default: Machine booted and ready!
[default] No installation found.
Loaded plugins: axelget, fastestmirror
Repository base is listed more than once in the configuration
Repository updates is listed more than once in the configuration
Repository extras is listed more than once in the configuration
Repository centosplus is listed more than once in the configuration
No metadata available for WANdisco-git
No metadata available for base
No metadata available for elrepo
No metadata available for elrepo-extras
No metadata available for elrepo-kernel
No metadata available for epel
No metadata available for extras
No metadata available for nginx
No metadata available for pgdg10
No metadata available for updates
Loading mirror speeds from cached hostfile
 * base: mirrors.aliyun.com
 * epel: mirrors.aliyun.com
 * extras: mirrors.shu.edu.cn
 * updates: mirrors.huaweicloud.com
Package kernel-devel-3.10.0-862.11.6.el7.x86_64 already installed and latest version
No package kernel-devel-4.18.7-1.el7.elrepo.x86_64 available.
Package gcc-4.8.5-28.el7_5.1.x86_64 already installed and latest version
Package binutils-2.27-28.base.el7_5.1.x86_64 already installed and latest version
Package 1:make-3.82-23.el7.x86_64 already installed and latest version
Package 4:perl-5.16.3-292.el7.x86_64 already installed and latest version
Package bzip2-1.0.6-13.el7.x86_64 already installed and latest version
Nothing to do
Copy iso file C:\Program Files\Oracle\VirtualBox\VBoxGuestAdditions.iso into the box /tmp/VBoxGuestAdditions.iso
Mounting Virtualbox Guest Additions ISO to: /mnt
mount: /dev/loop0 is write-protected, mounting read-only
Installing Virtualbox Guest Additions 5.2.18 - guest version is unknown
Verifying archive integrity... All good.
Uncompressing VirtualBox 5.2.18 Guest Additions for Linux........
VirtualBox Guest Additions installer
Removing installed version 5.2.18 of VirtualBox Guest Additions...
Copying additional installer modules ...
Installing additional modules ...
VirtualBox Guest Additions: Building the VirtualBox Guest Additions kernel modules.  This may take a while.
This system is currently not set up to build kernel modules.
Please install the Linux kernel "header" files matching the current kernel
for adding new hardware support to the system.
The distribution packages containing the headers are probably:
    kernel-devel kernel-devel-4.18.7-1.el7.elrepo.x86_64
VirtualBox Guest Additions: Starting.
VirtualBox Guest Additions: Building the VirtualBox Guest Additions kernel modules.  This may take a while.
This system is currently not set up to build kernel modules.
Please install the Linux kernel "header" files matching the current kernel
for adding new hardware support to the system.
The distribution packages containing the headers are probably:
    kernel-devel kernel-devel-4.18.7-1.el7.elrepo.x86_64
An error occurred during installation of VirtualBox Guest Additions 5.2.18. Some functionality may not work as intended.
In most cases it is OK that the "Window System drivers" installation failed.
Redirecting to /bin/systemctl start vboxadd.service
Job for vboxadd.service failed because the control process exited with error code. See "systemctl status vboxadd.service" and "journalctl -xe" for details.
Unmounting Virtualbox Guest Additions ISO from: /mnt
==> default: Checking for guest additions in VM...
    default: No guest additions were detected on the base box for this VM! Guest
    default: additions are required for forwarded ports, shared folders, host only
    default: networking, and more. If SSH fails on this machine, please install
    default: the guest additions and repackage the box to continue.
    default:
    default: This is not an error message; everything may continue to work properly,
    default: in which case you may ignore this message.
==> default: Setting hostname...
==> default: Configuring and enabling network interfaces...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: root
    default: SSH auth method: password
==> default: Rsyncing folder: /cygdrive/d/vagrant_env/dev/ => /vagrant
==> default: Mounting shared folders...
    default: /var/www => D:/wnmp/www
Vagrant was unable to mount VirtualBox shared folders. This is usually
because the filesystem "vboxsf" is not available. This filesystem is
made available via the VirtualBox Guest Additions and kernel module.
Please verify that these guest additions are properly installed in the
guest. This is not a bug in Vagrant and is usually caused by a faulty
Vagrant box. For context, the command attempted was:

mount -t vboxsf -o uid=0,gid=0 var_www /var/www

The error output from the command was:

/sbin/mount.vboxsf: mounting failed with the error: No such device
```

I can not find  `kernel-devel-4.18.7-1.el7.elrepo.x86_64` from  http://mirror.centos.org/centos/7.5.1804/updates/x86_64/Packages/ ,where can I find it ?

**Comment by fnordfish** [comment]
Hi,
I'm sorry but you ran into a common CentOS issue, where they won't provide kernel header/devel packages for "older" kernels.

TL;DR: The easiest way is to update your kernel and then install the header/devel package

You may want to try https://github.com/fnordfish/vagrant-vbguest-centos (not guaranteed to work all the time) and or read up on the issue:
- https://github.com/dotless-de/vagrant-vbguest/pull/231
- https://github.com/dotless-de/vagrant-vbguest/issues/223
- https://github.com/dotless-de/vagrant-vbguest/pull/153
- https://github.com/dotless-de/vagrant-vbguest/issues/141
